### PR TITLE
Compatibility with PHP 7 strict mode

### PR DESCRIPTION
**Pull request by kukulich**

### CODE REVIEW

File: src/Utils/Image.php [hunk 1]
```diff
@@ -321,15 +321,15 @@ public function resize($width, $height, $flags = self::FIT)
 	 */
 	public static function calculateSize($srcWidth, $srcHeight, $newWidth, $newHeight, $flags = self::FIT)
 	{
-		if (substr($newWidth, -1) === '%') {
-			$newWidth = round($srcWidth / 100 * abs($newWidth));
+		if (substr((string) $newWidth, -1) === '%') {
```
Comment (JanTvrdik) [inline src/Utils/Image.php]: Maybe `is_string($newWidth) && substr(...)`?

File: src/Utils/Image.php [hunk 2]
```diff
@@ -321,15 +321,15 @@ public function resize($width, $height, $flags = self::FIT)
...
+			$newWidth = (int) round($srcWidth / 100 * abs($newWidth));
 			$percents = TRUE;
 		} else {
 			$newWidth = (int) abs($newWidth);
 		}
 
-		if (substr($newHeight, -1) === '%') {
-			$newHeight = round($srcHeight / 100 * abs($newHeight));
+		if (substr((string) $newHeight, -1) === '%') {
+			$newHeight = (int) round($srcHeight / 100 * abs($newHeight));
```
Comment (JanTvrdik) [inline src/Utils/Image.php]: Casting `round` to int? Why not just cast the expression, i.e. `(int) ($srcHeight / 100 * abs($newHeight))`?

File: src/Utils/Strings.php [hunk 3]
```diff
@@ -205,7 +205,7 @@ public static function webalize($s, $charlist = NULL, $lower = TRUE)
 		if ($lower) {
 			$s = strtolower($s);
 		}
-		$s = preg_replace('#[^a-z0-9' . preg_quote($charlist, '#') . ']+#i', '-', $s);
+		$s = preg_replace('#[^a-z0-9' . preg_quote((string) $charlist, '#') . ']+#i', '-', $s);
```
Comment (JanTvrdik) [inline src/Utils/Strings.php]: Maybe avoid calling `preg_quote` for `NULL` value?

File: src/Utils/Image.php (hunk 2 above)
Comment (kukulich) [inline src/Utils/Image.php]: `(int) 1.9` would become `1` then so it's better to round it first.

Comment (JanTvrdik) [inline src/Utils/Image.php]: Maybe `(int) ($srcHeight / 100 * abs($newHeight) + 0.5)`?

File: src/Utils/Strings.php (hunk 3 above)
Comment (kukulich) [inline src/Utils/Strings.php]: Make sense, I will modify it.

File: src/Utils/Image.php (hunk 1 above)
Comment (kukulich) [inline src/Utils/Image.php]: Yes, `is_string` is probably better.

File: src/Utils/Image.php (hunk 2 above)
Comment (kukulich) [inline src/Utils/Image.php]: It's more readable with `round` for me. So it's up to @dg :)

Comment (dg) [inline src/Utils/Image.php]: round() is more readable and works with negative numbers, which are used by some methods in class Image. (ie. round(-1.6) === -2).

**Comment by kukulich** [comment]
Forced pushed so it's ready to merge I think.

File: src/Utils/FileSystem.php [hunk 4]
```diff
@@ -78,8 +78,8 @@ public static function delete($path)
 			}
 
 		} elseif (is_dir($path)) {
-			foreach (new \FilesystemIterator($path) as $item) {
-				static::delete($item);
+			foreach (new \FilesystemIterator($path) as $itemPath => $item) {
+				static::delete($itemPath);
```
Comment (dg) [inline src/Utils/FileSystem.php]: What about to use `(string) $itemPath`? To save one variable?

Comment (kukulich) [inline src/Utils/FileSystem.php]: Or `static::delete($item->getPathname());` ?

Comment (dg) [inline src/Utils/FileSystem.php]: Yes.

Comment (kukulich) [inline src/Utils/FileSystem.php]: Force pushed.

**Comment by dg** [comment]
Great, thank you!

**Pull request merged**